src/Microsoft.Health.Fhir.Liquid.Converter/Filters/DateFilters.cs

```diff
@@ -95,5 +95,30 @@ public static string Now(string input, string format = "yyyy-MM-ddTHH:mm:ss.FFFZ
         {
             return DateTime.UtcNow.ToString(format);
         }
+
+        public static string FormatAsHl7v2DateTime(string input, string timeZoneHandling = "preserve")
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return input;
+            }
+
+            if (!Enum.TryParse(timeZoneHandling, true, out TimeZoneHandlingMethod outputTimeZoneHandling))
+            {
+                throw new RenderException(FhirConverterErrorCode.InvalidTimeZoneHandling, Resources.InvalidTimeZoneHandling);
+            }
```

Comment:
nit: This block (before line 121) seems to be common across the other 3 functions too (with the datatimetype being diff for 1 of them) - can we pull it out into a common method.